### PR TITLE
Upgrade to protobuf 3.20.3

### PR DESCRIPTION
**Pull request by hamptonm1**
I upgraded Protobuf from 3.18.3 to 3.20.3 which matches what the `onnx` community currently has for version `1.13.0`. However, MacOS fails for version 3.20.3 (see this link for the reason: https://github.com/onnx/onnx-mlir/issues/1936). So what I have done based on @chentong319 suggestion, I upgraded everything else to 3.20.3 and kept Mac build on 3.18.3. Please let me know your thoughts.

### CODE REVIEW

File: docker/Dockerfile.llvm-project [hunk 1]
```diff
@@ -22,49 +22,49 @@ WORKDIR ${WORK_DIR}
...
+    make -j${NPROC} install && \
+    cd ${WORK_DIR} && rm -rf ninja; \
+    fi \
     && ln -sf /usr/bin/python3 /usr/bin/python
 
 # Install protobuf
-ARG PROTOBUF_VERSION=3.18.3
-RUN git clone -b v${PROTOBUF_VERSION} --recursive https://github.com/google/protobuf.git \
+ARG PROTOBUF_VERSION=3.20.3
+RUN git clone -b v${PROTOBUF_VERSION} --recursive https://github.com/protocolbuffers/protobuf.git \
```
Comment (gongsu832) [inline docker/Dockerfile.llvm-project:67]: This should be the only change to Dockerfile.llvm-project. The other messed up indentation changes should be reverted.

Comment (gongsu832) [reply docker/Dockerfile.llvm-project:67]: I meant the version and url changes should be the only changes needed.

Comment (hamptonm1) [reply docker/Dockerfile.llvm-project:67]: > I meant the version and url changes should be the only changes needed.

Sure!! Let me change the format

Comment (hamptonm1) [reply docker/Dockerfile.llvm-project:67]: @gongsu832 How does this look now? Ready to merge this in :)

**Review by gongsu832 — APPROVED** [review]
LGTM

File: utils/install-protobuf.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Check out protobuf v3.18.3
 PROTOBUF_VERSION=3.18.3
```
Comment (sstamenova) [inline utils/install-protobuf.sh:2]: So is this only used for Mac?

Comment (hamptonm1) [reply utils/install-protobuf.sh:2]: @sstamenova Yes!!! Only because Protobuf 3.20.3 failed for Mac, so kept it at the older version.

Comment (hamptonm1) [reply utils/install-protobuf.sh:2]: @sstamenova I explain why Mac was failing with Protobuf 3.20.3 here: https://github.com/onnx/onnx-mlir/issues/1936

Comment (sstamenova) [reply utils/install-protobuf.sh:2]: Thanks!

**Pull request merged**